src/main/java/org/apache/zeppelin/interpreter/MemoryListenerImpl.java

```diff
@@ -26,11 +26,14 @@
 
   @Override
   public void onEvent(MemoryEvent event) {
-
-    logger.warn("Critical Up memory event Occured");
-
     if (event.getState().isCritical()) {
+      logger.warn("Critical Up memory event occured");
       SnappyDataZeppelinInterpreter.cancelAllJobsAndPause();
+    } else if (event.getState().isNormal()) {
```

Comment:
This need not be a check for event.getState().isNormal(). Just if it is not critical and pauseInterpreter is already true then set it to false.